### PR TITLE
chore: version bump for taxonomy-connector.

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**

### CODE REVIEW

File: course_discovery/apps/course_metadata/tests/test_algolia_models.py [hunk 1]
```diff
@@ -442,9 +442,9 @@ def test_program_not_available_if_no_published_runs(self):
         assert program.availability_level == []
 
     def test_only_programs_with_spanish_courses_promoted_in_spanish_index(self):
-        all_spanish_program = AlgoliaProxyProgramFactory(partner=self.__class__.edxPartner)
-        mixed_language_program = AlgoliaProxyProgramFactory(partner=self.__class__.edxPartner)
-        all_english_program = AlgoliaProxyProgramFactory(partner=self.__class__.edxPartner)
+        all_spanish_program = AlgoliaProxyProgramFactory(partner=self.__class__.edxPartner, language_override=None)
+        mixed_language_program = AlgoliaProxyProgramFactory(partner=self.__class__.edxPartner, language_override=None)
+        all_english_program = AlgoliaProxyProgramFactory(partner=self.__class__.edxPartner, language_override=None)
```
Comment (HammadAhmadWaqas) [inline course_discovery/apps/course_metadata/tests/test_algolia_models.py:447]: Fixed this flaky test that was introduced due to changes in [promoted_in_spanish_index](https://github.com/openedx/course-discovery/commit/76b6e82097a71139948bc75bf206e89f79893e32#diff-1fc91f72e5dc28a966b180a151e35ea3bcc6f85236881faa99492c05ec38067aR505) method.

File: requirements/constraints.txt [hunk 2]
```diff
@@ -36,5 +36,5 @@ semgrep==0.102.0
 # Version 8.3.0 introduced changes that caused users to see User Disabled error
 edx-drf-extensions < 8.3.0
 
-# pinning because of 'cryptography' dependency conflict while installing snowflake-connector-python
-pyopenssl==22.0.0
```
Comment (Ali-D-Akbar) [inline requirements/constraints.txt:40]: Hmm, why did we remove this constraint?

**Review by Ali-D-Akbar — COMMENTED** [review]
Just one question before approval.

File: requirements/constraints.txt (hunk 2 above)
Comment (HammadAhmadWaqas) [reply requirements/constraints.txt:40]: pyopenssl==22.0.0 was causing problems while installing requirements on PR. Removing this constraint fix that problem. As we are not getting any snowflake-connector-python problem after removing it, It should be a good thing to remove a contraint.

File: requirements/constraints.txt [hunk 3]
```diff
@@ -36,5 +36,5 @@ semgrep==0.102.0
...
+# Pinning sphinx as new major release is causing error while running on
```
Comment (DawoudSheraz) [inline requirements/constraints.txt]: while running on?

**Review by DawoudSheraz — APPROVED** [review]
To update the comment in constraints.txt

**Review by Ali-D-Akbar — APPROVED** [review]

**Comment by HammadAhmadWaqas** [comment]
@Ali-D-Akbar @DawoudSheraz I have updated the comment. Could you please merge it?

**Pull request merged**